tests/unit/test_pybamm_data.py

```diff
@@ -0,0 +1,37 @@
+import pybamm
+import pytest
+import random
+from tests import no_internet_connection
+
+data_loader = pybamm.DataLoader()
+
+
+@pytest.mark.skipif(
+    no_internet_connection(),
+    reason="Network not available to download files from registry",
+)  # Skip if no internet
+def test_fetch():
+    # Fetch a file from the registry and check it in local cache folder for its presence
+    data_loader = pybamm.DataLoader()
+    random_file = random.choice(list(data_loader.files.keys()))
+    file_path = data_loader.get_data(random_file)
+    return file_path.is_file()
```

Comment:
I agree with codacy here, random in tests is not great. Can we just check the first file instead?
```suggestion
def test_fetch():
    test_file = list(pybamm.DataLoader().files.keys())[0]
    return data_loader.get_data(test_file).is_file()
```